### ISSUE TITLE
Different results on different machines

### ISSUE THREAD

**Issue by horsepurve** [apache/incubator-mxnet#4479]
Hi, I run lstm regression model using mxnet on different machines, and get different results, specifically, I run on 3 machines and one of which is always worse than the other two:

1. CentOS 6.5 64 bit + gcc 4.8.1 + openblas + mxnet 0.7/0.9 = pearson correlation 0.96+
2. Windows 10 32 bit + vs 2015 + openblas/mkl + mxnet 0.7 = pearson correlation 0.96+
3. fedora 25/ubuntu 16.10 32 bit + gcc 6.3/clang 3.8 + openblas + mxnet 0.7/0.9 = pearson correlation 0.90~0.93

The data, scripts etc. on 3 machines are just the same, but the result on the third machine is much worse than others, can anybody please help? Thanks.

**Comment by piiswrong** [comment]
could you run the test suite on fedora? see tests/jenkins/run_test.sh

I suspect gcc6.3 might be the cause

**Comment by xuancong84** [comment]
What if you train a model on one machine and test it on another machine (with all dropouts set to 1 if you have used dropout)? Only if this test gives significantly different result (beyond machine precision level), then it can confirm the presence of a bug.

**Comment by horsepurve** [comment]
@piiswrong thanks for your advice. I changed to gcc 4.8.1 and the problem remains. Then I run some testings, finding that mxnet on the 3rd machine cannot even pass tests/python/train/test_mlp.py due to the accuracy of merely 0.90. However, if I change batch size from 100 to 32 and learning rate from 0.1 to 0.01, then it can pass test_mlp.py. I cannot figure out why mxnet behaves differently on this machine.

**Comment by piiswrong** [comment]
does it pass operator test?

**Comment by horsepurve** [comment]
It cannot pass two of them: test_deconvolution() and test_convolution_dilated_impulse_response(), moreover, the function check_deconvolution_forward_backward() gives different results every time I run it, sometimes it passes, sometime it doesn't pass (out != args_grad[0].asnumpy()), I'm totally freaking out.

**Comment by piiswrong** [comment]
are you using cuda? If not your CPU is probably failing...

**Comment by piiswrong** [comment]
Or memory...

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
